### PR TITLE
hstr: update to 2.5

### PR DESCRIPTION
**Pull request by NeelChotai**
#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

**Comment by paper42** [comment]
Can we do something simpler in pre_configure or is there a reason why it's done this way?
```sh
pre_configure() {
    vsed -i 's|ncursesw/curses.h|curses.h|g' src/include/hstr.h src/include/hstr_curses.h
	autoreconf -fi
}
```

**Comment by NeelChotai** [comment]
> Can we do something simpler in pre_configure or is there a reason why it's done this way?
> 
> ```shell
> pre_configure() {
>     vsed -i 's|ncursesw/curses.h|curses.h|g' src/include/hstr.h src/include/hstr_curses.h
> 	autoreconf -fi
> }
> ```

Doesn't seem to be any particular reason, updated!

File: srcpkgs/hstr/template [hunk 1]
```diff
@@ -1,22 +1,19 @@
 # Template file for 'hstr'
 pkgname=hstr
-version=2.3
+version=2.5
 revision=1
 build_style=gnu-configure
-hostmakedepends="automake pkg-config"
+hostmakedepends="autoconf automake pkg-config"
```
Comment (paper42) [inline srcpkgs/hstr/template:6]: autoconf is already a dependency of automake

Comment (NeelChotai) [reply srcpkgs/hstr/template]: fixed!

**Pull request merged**